dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SecretTool.java

```diff
@@ -51,12 +74,18 @@ public Object getSystemSecret (final String key) {
 	public Object getSystemSecret (final String key,
 								   final Object defaultValue) {
 
+        if(!canUserEvaluate()){
+            throw new SecurityException("External scripting is disabled in your dotcms instance.");
+        }
 		final Optional<DotVelocitySecretAppConfig> config = DotVelocitySecretAppConfig.config(APILocator.systemHost());
 		return config.isPresent()? config.get().getStringOrNull(key, null!= defaultValue? defaultValue.toString():null) : defaultValue;
 	}
 
 	public char[] getCharArray(final String key) {
 
+        if(!canUserEvaluate()){
+            throw new SecurityException("External scripting is disabled in your dotcms instance.");
```

Comment:
Just throw a SecurityException once in the `canUserEvaluate` and get rid of all these if statements